onchaind/onchaind.c

```diff
@@ -3640,6 +3647,12 @@ static void handle_unknown_commitment(const struct tx_parts *tx,
 	wait_for_resolved(outs);
 }
 
+static int cmp_htlc_cltv(const struct htlc_stub *a,
+			 const struct htlc_stub *b, void *unused)
+{
+	return a->cltv_expiry - b->cltv_expiry;
```

Comment:
`cltv_expiry` is unsigned 32-bit.  If `int` is not 32-bit, this might not work correctly (it might not sign-extend if `int` is 64-bit).  I suggest casting to `int` before doing the subtraction, or better yet, just being explicit about the compare --- on most CPUs the cost of a subtract is the same as the cost of a compare (some use the same instruction for it!), and most compilers can compile `if (a < b) return -1; else if (a == b) return 0; else return 1;` into a compare/subtract instruction and extraction from flags register.

Reply:
Yes, will fix.

Good catch!